src/errors/InvalidProfileParentError.ts

```diff
@@ -0,0 +1,8 @@
+import { WithSource } from './WithSource';
+import { SourceInfo } from '../fshtypes';
+
+export class InvalidProfileParentError extends Error implements WithSource {
+  constructor(public childName: string, public parentName: string, public sourceInfo: SourceInfo) {
+    super(`Parent ${parentName} is not a valid Parent for Profile ${childName}.`);
```

Comment:
Here I am also not sure about this. Why _can't_ a profile just inherit from a logical model. I guess this isn't something we currently support in SUSHI, because we don't add logical models to our `FHIRDefinitions`, so you couldn't inherit from one. But I'm not able to figure out why we need to prohibit this.

Reply:
Another good question. This should be included in the general discussion on allowed parents.

Reply:
If a profile inherits from a logical model, it should result in a logical model having a `derivation = 'constraint'` - essentially "profiling" the base logical model. Rather than trying to fit that in now, I recommend that it be deferred until we can get feedback from the community with actual use cases for "profiling logical models".